### PR TITLE
ory-hydra: init at 2.0.3

### PR DESCRIPTION
**Pull request by arianvp**
###### Description of changes

OpenID Certified™ OpenID Connect and OAuth Provider written in Go - cloud native, security-first, open source API security for your infrastructure. SDKs for any language. Works with Hardware Security Modules. Compatible with MITREid.

https://github.com/ory/hydra

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by arianvp** [comment]
A lot of their tests shell out to docker. I'll try to cherry-pick the ones that work without so we have at least some test coverage

**Comment by arianvp** [comment]
@GrahamcOfBorg build ora-hydra

**Comment by arianvp** [comment]
@GrahamcOfBorg build ory-hydra

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -39645,4 +39645,6 @@ with pkgs;
   udict = callPackage ../applications/misc/udict { };
 
   duden = callPackage ../applications/misc/duden { };
+
+  ory-hydra = callPackage ../servers/ory-hydra { };
```
Comment (Artturin) [inline pkgs/top-level/all-packages.nix:39649]: wrong location

**Comment by Artturin** [comment]
Please use your own fork for your pr branches

**Comment by arianvp** [comment]
> Please use your own fork for your pr branches

Is this a documented convention?  Wasn't ware of it

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (arianvp) [reply pkgs/top-level/all-packages.nix:39649]: What is the correct location?

**Comment by Artturin** [comment]
> > Please use your own fork for your pr branches
> 
> Is this a documented convention? Wasn't ware of it

not in nixpkgs but AFAIK it's basic github hygiene

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (arianvp) [reply pkgs/top-level/all-packages.nix:39649]: Thanks! Opening a new pr soon

**Pull request closed without merging**